### PR TITLE
Karma unit test, with watch and CI single run

### PR DESCRIPTION
**Pull request by settinghead**
- There is no gulp process yet, as there isn't a need for one
- run `npm install` and `npm run test` to test if karma is working

### CODE REVIEW

**Comment by settinghead** [comment]
@alex-deaconu @ezequielc please review.

A couple of notes:

- Although the karma test config file shares a lot in common with the one in `widget-tester`, I think the purpose for `widget-tester` is too broad at this point and it would hinder newer versions and features from being introduced to our testing structure for this project. Therefore, after weighing the tradeoff, I decided to create a separate `karma.config.js` instead of reusing `widget-tester`.
- I have been delaying the introduction of gulp and am opting for just running the karma binary currently (see [code here](https://github.com/Rise-Vision/viewer/pull/107/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R7). Like I said in the discussion, gulp is a good control flow tool, but it also introduces new bugs and masks the visibility of errors in many cases. I'd much prefer if we can do without it (and instead just open up a few terminals and run single-purpose commands). Let me know if you have any thoughts on this.

Cheers.

**Comment by alex-deaconu** [comment]
@settinghead @ezequielc I don't really have an opinion with `.test.js` files being included with the source code. Also OK with the separate `karma.config.js` file. 

However, I would like to get `coveralls` integrated (maybe not now, but at some point). Do you think we can do that without `widget-tester` (and `gulp`)?

Thanks.

**Comment by settinghead** [comment]
@alex-deaconu Sure I'll add Coverall later. We probably don't need `widget-tester`, but not sure if we need gulp for it, but if it's the case, we'll have to add it.

**Comment by alex-deaconu** [comment]
@settinghead sounds good 👍 . Thanks!

File: README.md [hunk 1]
```diff
@@ -7,11 +7,40 @@ Viewer renders HTML content with the Chrome / Chromium browser on displays. Typi
...
+### With watch
+
+```bash
+npm run test
+```
+
+### Single run for Continuous Integration
+
+```bash
+npm run test ci
```
Comment (alex-deaconu) [inline README.md]: @settinghead small typo here, I think the command should be `npm run test:ci`. Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc @settinghead as Xiyang is on vacation I will take over this PR. Will fix typo and move the JS files to the Viewer's `webapp` folder (https://github.com/Rise-Vision/viewer/tree/master/src/main/webapp) so they get deployed correctly. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu Sounds good to me. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc PR should be ready to be merged. Can you please review? Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu Looks good 👍 
Just to be 100% sure, do you think it would be worth to make the test fail and verify that it does not deploy in that case? Cheers!

**Comment by alex-deaconu** [comment]
@ezequielc good call. Added a bad commit and the tests failed: no deployment (https://circleci.com/gh/Rise-Vision/viewer/250). Also squashed, I think we're good to go. Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Nice. Thanks for checking it 👍

**Pull request merged**